src/server/routes/collection.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2020 Prabal Singh
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as auth from '../helpers/auth';
+import * as middleware from '../helpers/middleware';
+import * as propHelpers from '../../client/helpers/props';
+import {escapeProps, generateProps} from '../helpers/props';
+import Layout from '../../client/containers/layout';
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
+import UserCollectionForm from '../../client/components/forms/userCollection';
+import express from 'express';
+import {makeCollectionCreateOrEditHandler} from '../helpers/collectionRouteUtils';
+import target from '../templates/target';
+
+
+const router = express.Router();
+
+/* If the route specifies a CollectionId, load the Collection for it. */
+router.param(
+	'collectionId',
+	middleware.makeCollectionLoader()
+);
```

Comment:
You can move this after the /create route; that way you're sure it only looks for collectionID parameter in the routes that need it (I recently changes this in the entity routes in PR #429 )